charts/opentelemetry-collector/templates/config/_metrics.tpl

```diff
@@ -1,3 +1,10 @@
+{{/*
+Returns true if any config override defines a metrics pipeline.
+*/}}
+{{- define "opentelemetry-collector.metrics.enabled" -}}
+{{- or .Values.configOverride.service.pipelines.metrics  .Values.agentCollector.configOverride.service.pipelines.metrics .Values.standaloneCollector.configOverride.service.pipelines.metrics }}
```

Comment:
This would require all of the `configOverride` sources to be defined up to `metrics` key, otherwise it causes `nil pointer evaluating interface {}.pipelines`.

We could use `.Values.configOverride.service.pipelines.metrics` only to disable a pipeline.

Another option is to have all the parts of the pipelines (receivers+processors+exporters, not only `exporters`) pre-defined in `values.yaml` ->`.Values.configOverride` instead of `_metrics.tpl`. In that case it will be fully overridable, and a pipeline can be easily disabled that way.

Reply:
Good point and I'll try the latter approach of moving more to the values file

Reply:
Alright I've moved the metrics and traces bases straight into values, and have renamed configOverride to config, although it is technically an override, what it overrides is pretty minimal, WDYT?

Reply:
And I just converged all config related helpers into `_config.tpl` if that's alright?

Reply:
resulting out of the box config is

```yaml
data:
  relay: |
    exporters:
      logging: {}
    extensions:
      health_check: {}
    processors:
      batch: {}
      memory_limiter:
        ballast_size_mib: 204
        check_interval: 5s
        limit_mib: 409
        spike_limit_mib: 128
      queued_retry: {}
    receivers:
      jaeger:
        protocols:
          grpc:
            endpoint: 0.0.0.0:14250
          thrift_http:
            endpoint: 0.0.0.0:14268
      otlp:
        protocols:
          grpc: null
          http: null
      prometheus:
        config:
          scrape_configs:
          - job_name: opentelemetry-collector
            scrape_interval: 10s
            static_configs:
            - targets:
              - ${MY_POD_IP}:8888
      zipkin:
        endpoint: 0.0.0.0:9411
    service:
      extensions:
      - health_check
      pipelines:
        metrics:
          exporters:
          - logging
          processors:
          - memory_limiter
          - batch
          - queued_retry]
          receivers:
          - otlp
          - prometheus
        traces:
          exporters:
          - logging
          processors:
          - memory_limiter
          - batch
          - queued_retry
          receivers:
          - otlp
          - jaeger
          - zipkin
```


And setting metrics and traces to null yields:

```yaml
    exporters:
      logging: {}
    extensions:
      health_check: {}
    processors:
      batch: {}
      memory_limiter:
        ballast_size_mib: 204
        check_interval: 5s
        limit_mib: 409
        spike_limit_mib: 128
      queued_retry: {}
    receivers:
      jaeger:
        protocols:
          grpc:
            endpoint: 0.0.0.0:14250
          thrift_http:
            endpoint: 0.0.0.0:14268
      otlp:
        protocols:
          grpc: null
          http: null
      prometheus:
        config:
          scrape_configs:
          - job_name: opentelemetry-collector
            scrape_interval: 10s
            static_configs:
            - targets:
              - ${MY_POD_IP}:8888
      zipkin:
        endpoint: 0.0.0.0:9411
    service:
      extensions:
      - health_check
      pipelines:
        metrics: null
        traces: null
```

Reply:
> And setting metrics and traces to null yields:

interesting. disabling a pipeline with
```
config:
  service:
    pipelines:
      metrics: null
```
should remove the key from pipelines, not setting it to `null`. That's exactly what I got when testing it on helm 3.4.0

Reply:
> Alright I've moved the metrics and traces bases straight into values, and have renamed configOverride to config, although it is technically an override, what it overrides is pretty minimal, WDYT?

`config` sounds good to me

Reply:
Yeah, that sounds like a good idea.